src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -694,6 +694,7 @@ def do_resolve_with_resolved_children(
                 if logical_plan.condition
                 else None,
                 resolved_children.get(logical_plan.source_data, None),
+                logical_plan,
```

Comment:
I searched in the source code but didn't figure out how the `source_plan` impact the finally generated SQL. Could you point me to that place?

Reply:
Actually it's not used anywhere (just because scala code has it), and I added it here to just keep consistent with other methods in this file.